test/manipulation.test.js

```diff
@@ -902,6 +913,72 @@ describe('manipulation', function() {
     });
   });
 
+  bdd.describeIf(connectorCapabilities.supportForceId !== false,
+    'updateOrCreate when forceId is true', function() {
+      var Post;
+      before(function definePostModel(done) {
+        var ds = getSchema();
+        Post = ds.define('Post', {
+          title: {type: String, length: 255},
+          content: {type: String},
+        }, {forceId: true});
+        ds.automigrate('Post', done);
+      });
+
+      it('fails when id does not exist in db & validate is true', function(done) {
+        var unknownId = uid.fromConnector(db) || 123;
+        var post = {id: unknownId, title: 'a', content: 'AAA'};
+        Post.updateOrCreate(post, {validate: true}, (err) => {
+          err.statusCode.should.equal(404);
```

Comment:
This will fail with `TypeError` when `updateOrCreate` passes and `err` is undefined.

Here is better assert:

```js
should(err).have.property('statusCode', 404);
```